src/events/embedded/types.ts

```diff
@@ -1,116 +1,81 @@
-export interface IEmbeddedMessageElementsButton {
+interface Session {
   id: string;
-  title: string;
-  action?: IEmbeddedMessageElementsButtonAction;
+  start?: number;
+  end?: number;
 }
-export interface IEmbeddedMessageElementsButtonAction {
+
+interface Impression {
+  messageId: string;
+  displayCount: number;
+  displayDuration: number;
+}
+
+interface ButtonAction {
   type: string;
   data?: string;
 }
-export interface IEmbeddedMessageElements {
+
+interface Button {
+  id: string;
+  title: string;
+  action?: ButtonAction;
+}
+
+interface Elements {
   title?: string;
   body?: string;
   mediaUrl?: string;
 
-  buttons?: [IEmbeddedMessageElementsButton];
-  text?: [IEmbeddedMessageElementsText];
-  defaultAction?: IEmbeddedMessageElementsDefaultAction;
+  buttons?: Button[];
+  text?: Text[];
+  defaultAction?: DefaultAction;
```

Comment:
Similar comment here. Even when set to "none" there's a value we pass in for that, right?